front/src/app/components/forms/establishment/highlightStringsFromMatches.ts

```diff
@@ -0,0 +1,71 @@
+import { MatchRangeDto } from "shared";
+
+export type SliceOfString = {
+  startIndexInclusive: number;
+  endIndexExclusive: number;
+  bolded: boolean;
+};
+
+export const highlightStringsFromMatches = (
+  matchRanges: MatchRangeDto[],
+  description: string,
+): SliceOfString[] => {
+  const result: SliceOfString[] = [];
+  const matchRangesWithoutOverlaps = mergeOverlappingMatchRanges(
+    matchRanges,
+    description.length,
+  );
+  let prevEndIndex = 0;
+
+  for (let i = 0; i < matchRangesWithoutOverlaps.length; i++) {
```

Comment:
Est-ce que tu penses qu'on pourrait se passer des boucles for et des result.push() ?